### PR TITLE
fix markdown page script injection

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Markdown script injection was broken in a couple of small ways.
- This PR hardens our script injection to use TS types better.
- This PR also adds a Markdown example page to the with-tailwindcss example project
- Last, we add a test to the test suite and make some small improvements to the tailwind test suite.

## Testing

- Manually confirmed and tests added.

## Docs

- N/A

### CODE REVIEW

File: packages/astro/test/fixtures/tailwindcss/src/pages/index.astro [hunk 1]
```diff
@@ -2,11 +2,6 @@
 // Component Imports
 import Button from '../components/Button.astro';
 import Complex from '../components/Complex.astro';
-
-import "../styles/global.css";
-
-// Full Astro Component Syntax:
-// https://docs.astro.build/core-concepts/astro-components/
```
Comment (natemoo-re) [inline packages/astro/test/fixtures/tailwindcss/src/pages/index.astro:9]: Love it!

File: packages/astro/test/tailwindcss.test.js [hunk 2]
```diff
@@ -55,6 +55,14 @@ describe('Tailwind', () => {
...
 			expect(button.hasClass('2xl:w-[80%]'), 'complex class').to.be.true;
 		});
+
+		it('handles Markdown pages', async () => {
+			const html = await fixture.readFile('/markdown-page/index.html');
+			const $ = cheerio.load(html);
+			const bundledCSSHREF = $('link[rel=stylesheet][href^=/assets/]').attr('href');
+			const bundledCSS = await fixture.readFile(bundledCSSHREF.replace(/^\/?/, '/'));
+			expect(bundledCSS, 'includes used component classes').to.match(/\.bg-purple-600{/);
+		});
```
Comment (natemoo-re) [inline packages/astro/test/tailwindcss.test.js:65]: Was just about to ask if this was covered in our suite ☺️

**Review by natemoo-re — APPROVED** [review]
Looks like a clean refactor and fix!

**Comment by FredKSchott** [comment]
FYI GH Actions are down but I believe tests are failing on a windows issue, will wait for them to come back up to debug

**Pull request merged**